src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -230,43 +397,46 @@ void DoorShutter_Init(Actor* thisx, PlayState* play2) {
     this->dyna.actor.home.pos.z = this->dyna.actor.shape.yOffset;
     DynaPolyActor_Init(&this->dyna, 0);
     this->doorType = (this->dyna.actor.params >> 6) & 0xF;
-    phi_a3 = D_80998224[this->doorType];
-    if (phi_a3 < 0) {
-        ShutterSceneInfo* phi_v1;
 
-        for (phi_v1 = &sSceneInfo[0], i = 0; i < ARRAY_COUNT(sSceneInfo) - 1; i++, phi_v1++) {
-            if (play->sceneId == phi_v1->sceneId) {
+    styleType = sTypeStyles[this->doorType];
+    if (styleType < 0) {
+        DoorShutterSceneInfo* sceneInfo;
+
+        for (sceneInfo = &sSceneInfo[0], i = 0; i < ARRAY_COUNT(sSceneInfo) - 1; i++, sceneInfo++) {
+            if (play->sceneId == sceneInfo->sceneId) {
                 break;
             }
         }
-        phi_a3 = phi_v1->index;
-    } else if (phi_a3 == 6) {
-        BossDoorInfo* phi_v1_2;
+        styleType = sceneInfo->styleType;
+    } else if (styleType == DOORSHUTTER_STYLE_BOSS_DOOR) {
+        DoorShutterBossDoorInfo* bossDoorInfo;
 
-        for (phi_v1_2 = &D_80998288[0], i = 0; i < ARRAY_COUNT(D_80998288) - 1; i++, phi_v1_2++) {
-            if (play->sceneId == phi_v1_2->dungeonSceneId || play->sceneId == phi_v1_2->bossSceneId) {
+        for (bossDoorInfo = &sBossDoorInfo[0], i = 0; i < ARRAY_COUNT(sBossDoorInfo) - 1; i++, bossDoorInfo++) {
+            if (play->sceneId == bossDoorInfo->dungeonSceneId || play->sceneId == bossDoorInfo->bossSceneId) {
                 break;
             }
         }
-        this->unk_168 = phi_v1_2->index;
+        this->bossDoorTexIndex = bossDoorInfo->texIndex;
     } else {
         this->dyna.actor.room = -1;
     }
-    if (this->requiredObjBankIndex = objectIndex = Object_GetIndex(&play->objectCtx, sObjectInfo[phi_a3].objectId),
+
+    if (this->requiredObjBankIndex = objectIndex = Object_GetIndex(&play->objectCtx, sStyleInfo[styleType].objectId),
         (s8)objectIndex < 0) {
         Actor_Kill(&this->dyna.actor);
         return;
     }
+
     DoorShutter_SetupAction(this, DoorShutter_SetupType);
```

Comment:
i kinda think this SetupType should be named after the usual WaitForObject scheme, and then the stuff it does when the object is ready is initialize the object related stuff. but idk